src/generator/AutoRest.Go/Templates/MethodTemplate.cshtml

```diff
@@ -229,3 +211,59 @@ func (client @(Model.Owner)) @(Model.ResponderMethodName)(resp *http.Response) (
     @:return
 @:}
 }
+
+@if (Model.IsPageable && !Model.IsNextMethod && Model.ListElement != null)
+{
+@:@EmptyLine
+@:// @(Model.ListCompleteMethodName) gets all elements from the list without paging.
+@:func (client @(Model.Owner)) @(Model.ListCompleteMethodName) @(Model.MethodParametersSignatureComplete) @(Model.MethodReturnSignatureComplete) {
+    @:resultChan := make(chan @((Model.ListElement.ModelType as SequenceTypeGo).GetElement))
+    @:errChan := make(chan error, 1)
+    @:go func() {
+        @:defer func() {
+			@:close(resultChan)
+			@:close(errChan)
+		@:}()
+        @:list, err := client.@(Model.Name)(@(Model.HelperInvocationParameters(false)))
+        @:if err != nil {
+            @:errChan <- err
+            @:return
+        @:}
+        @:if list.@(Model.ListElement.Name) != nil {
+            @:for _, item := range *list.@(Model.ListElement.Name) {
+                @:select {
+                @:case <- cancel:
+                    @:return
+                @:default:
+                    @:resultChan <- item
+                @:}
```

Comment:
As written, the above will never really consider cancellation. Instead of using "default" the code we generate should look like this:

``` Go
for _, item := range *list.Items {
    select {
    case <- cancel:
        return
    case resultChan <- item:
        // Intentionally Left Blank
    }
}
```